python/ray/data/block.py

```diff
@@ -276,6 +276,10 @@ def take(self, indices: List[int]) -> Block:
         """
         raise NotImplementedError
 
+    def select(self, keys: List[KeyFn]) -> Block:
```

Comment:
Why limit to keys? We're going to support column projection generally.

Reply:
@jianoaix - `KeyFn` can be a column (for Arrow and Pandas block), or function (for Python simple block). So we can use the same API to support column projection. `ds.select_columns(cols)` would just need to call `BlockAccessor.for_block(block).select(cols)`.

Reply:
Then it's better not calling "key" as that seems to indicate a mismatch what it intends to support.

Reply:
@jianoaix - does `columns` sounds good to you?

Reply:
Sounds good!

Reply:
@jianoaix - updated.